### ISSUE TITLE
Exception in oauth2/token api call

### ISSUE THREAD

**Issue by yogeshs1989** [filsh/yii2-oauth2-server#132]
Whenever I am calling oauth2/token api, I am getting following exception.

`Argument 1 passed to OAuth2\\Server::handleTokenRequest() must be an instance of OAuth2\\RequestInterface, instance of yii\\web\\Request given, called in /Applications/XAMPP/xamppfiles/htdocs/api_new/vendor/filsh/yii2-oauth2-server/Server.php on line 39`

Well I have resolved this bug by changing the code in `vendor/filsh/yii2-oauth2-server/Module.php` file

Earlier Code:-

> public function getRequest()
    {
        if(!$this->has('request')) {
            $this->set('request', Request::createFromGlobals());
        }
        return $this->get('request');
    }

Changed Code:-
> public function getRequest()
    {
        $this->set('request', \OAuth2\Request::createFromGlobals());
        return $this->get('request');
    }

**Comment by hiqsol** [comment]
I've met same problem. It comes from this commit:

https://github.com/yiisoft/yii2/commit/4081a4b20349974ea9f37957b166fa4398cf3a8b

Now yii Module gets components from parents by default.

But actual problem is that this Module shouldn't keep Request and Response as components.
Code should be something like this:

```php
    if ($this->_request === null) {
        $this->_request = Request::createFromGlobals();
    }
    return $this->_request;
```

**Comment by oisoftware** [comment]
@hiqsol @yogeshs1989 How did you resolve this issue. We are having the same problem

**Comment by hiqsol** [comment]
This is how I solved the problem through configuration.
It's a bit rude but requires no chages to this package.

```php
    'modules' => [ 
        'oauth2' => [ 
            'class' => \filsh\yii2\oauth2server\Module::class,
            'components' => [ 
                'request' => function () { 
                    return \filsh\yii2\oauth2server\Request::createFromGlobals();
                },
                'response' => [
                    'class' => \filsh\yii2\oauth2server\Response::class,
                ],
            ],
    ...
```

**Comment by yogeshs1989** [comment]
@oisoftware I have posted the answer how I have resolved this. I haven't tried @hiqsol's solution, but it looks promising.

**Comment by luciomr** [comment]
@hiqsol 's solution worked fine here. Tested on Yii 2.14.1 and filsh/yii2-oauth2-server 2.0.1

**Comment by zdhukui** [comment]
> Whenever I am calling oauth2/token api, I am getting following exception.
> 
> `Argument 1 passed to OAuth2\\Server::handleTokenRequest() must be an instance of OAuth2\\RequestInterface, instance of yii\\web\\Request given, called in /Applications/XAMPP/xamppfiles/htdocs/api_new/vendor/filsh/yii2-oauth2-server/Server.php on line 39`
> 
> Well I have resolved this bug by changing the code in `vendor/filsh/yii2-oauth2-server/Module.php` file
> 
> Earlier Code:-
> 
> > public function getRequest()
> > {
> > if(!$this->has('request')) {
> > $this->set('request', Request::createFromGlobals());
> > }
> > return $this->get('request');
> > }
> 
> Changed Code:-
> 
> > public function getRequest()
> > {
> > $this->set('request', \OAuth2\Request::createFromGlobals());
> > return $this->get('request');
> > }

It works! Thanks!

**Comment by zdhukui** [comment]
> This is how I solved the problem through configuration.
> It's a bit rude but requires no chages to this package.
> 
> ```
>     'modules' => [ 
>         'oauth2' => [ 
>             'class' => \filsh\yii2\oauth2server\Module::class,
>             'components' => [ 
>                 'request' => function () { 
>                     return \filsh\yii2\oauth2server\Request::createFromGlobals();
>                 },
>                 'response' => [
>                     'class' => \filsh\yii2\oauth2server\Response::class,
>                 ],
>             ],
>     ...
> ```

HOHO, good method, thanks
`        'oauth2' => [
            'class' => \filsh\yii2\oauth2server\Module::class,
            'tokenParamName' => 'accessToken',
            'tokenAccessLifetime' => 3600 * 24,
            'storageMap' => [
                'user_credentials' => 'common\models\User',
            ],
            'grantTypes' => [
                'user_credentials' => [
                    'class' => 'OAuth2\GrantType\UserCredentials',
                ],
                'refresh_token' => [
                    'class' => 'OAuth2\GrantType\RefreshToken',
                    'always_issue_new_refresh_token' => true
                ]
            ],
            'components' => [
                'request' => function () {
                    return \filsh\yii2\oauth2server\Request::createFromGlobals();
                },
                'response' => [
                    'class' => \filsh\yii2\oauth2server\Response::class,
                ],
            ],
        ],`

**Comment by ar1fbudiman** [comment]
> Whenever I am calling oauth2/token api, I am getting following exception.
> 
> `Argument 1 passed to OAuth2\\Server::handleTokenRequest() must be an instance of OAuth2\\RequestInterface, instance of yii\\web\\Request given, called in /Applications/XAMPP/xamppfiles/htdocs/api_new/vendor/filsh/yii2-oauth2-server/Server.php on line 39`
> 
> Well I have resolved this bug by changing the code in `vendor/filsh/yii2-oauth2-server/Module.php` file
> 
> Earlier Code:-
> 
> > public function getRequest()
> > {
> > if(!$this->has('request')) {
> > $this->set('request', Request::createFromGlobals());
> > }
> > return $this->get('request');
> > }
> 
> Changed Code:-
> 
> > public function getRequest()
> > {
> > $this->set('request', \OAuth2\Request::createFromGlobals());
> > return $this->get('request');
> > }

worked for me, thank you

**Comment by varp** [comment]
@hiqsol @yogeshs1989 @luciomr @oisoftware @zdhukui  Fixed in v2.1.0
